src/Streamly/Internal/Data/Fold.hs

```diff
@@ -550,6 +555,14 @@ defaultSalt = 0x087fc72c
 rollingHash :: (Monad m, Enum a) => Fold m a Int
 rollingHash = rollingHashWithSalt defaultSalt
 
+{-# INLINABLE rollingHashFirstN #-}
+rollingHashFirstN :: (Monad m, Enum a) => Int -> Fold m a Int
+rollingHashFirstN n = ltake n rollingHash
+
+{-# INLINABLE rollingHashLastN #-}
+rollingHashLastN :: (MonadIO m, Enum a, Storable a) => Int -> Fold m a Int
+rollingHashLastN n = lastN n rollingHash
```

Comment:
Granted I have to change the implementation of `lastN`, the current implementation would not recompute checksum every time. It will only compute once, only for the `lastN` elements. Current implementation of `lastN` is equivalant to `applyFoldLastN`.

Reply:
> if we use it as a scan it will be recomputed every time, it will be very expensive.
Missed this sentence somehow.